### PR TITLE
[breaking batch] remove the sign from integer literals in the ast

### PR DESCRIPTION
**Pull request by oli-obk**
r? @eddyb

### CODE REVIEW

File: src/libsyntax/ext/quote.rs [hunk 1]
```diff
@@ -263,9 +263,24 @@ pub mod rt {
...
+                    let lit = ast::LitInt(val as u64, ast::SignedIntLit($tag));
+                    let lit = P(ast::Expr {
+                        id: ast::DUMMY_NODE_ID,
+                        node: ast::ExprLit(P(dummy_spanned(lit))),
+                        span: DUMMY_SP,
+                        attrs: None,
+                    });
+                    P(ast::Expr {
+                        id: ast::DUMMY_NODE_ID,
+                        node: ast::ExprUnary(ast::UnNeg, lit),
```
Comment (eddyb) [inline src/libsyntax/ext/quote.rs]: Shouldn't `lit` be returned early if the value is positive?

Comment (oli-obk) [inline src/libsyntax/ext/quote.rs]: indeed. fixed it

**Comment by eddyb** [comment]
LGTM. r? @Manishearth for confirmal of acceptable breakage.

**Comment by Manishearth** [comment]
This will likely cause aster breakage. I'm mostly okay with the clippy breakage this may cause (there's not much using this)

**Comment by Manishearth** [comment]
cc @erickt @nrc 

We might want to batch this up with other syntax breaking changes

**Comment by nrc** [comment]
What's the motivation for this work?

Seems like a good change to make to the HIR, not sure if it is a good change for the AST - does this conflate different source situations? (`-5` vs `- 5` or anything).

At the least, this should be batched with other changes.

**Comment by eddyb** [comment]
@nrc Rust doesn't have `-5` literals and the AST was misrepresenting that fact.

**Comment by oli-obk** [comment]
The only way to get a `-5` literal was to create it in a compiler plugin. So I'm not even sure if the current trans code did what it was supposed to, but it was also hard to check whether it did. The HIR could potentially fold `ExprUnary(Neg, Lit)` into some kind of signed literal.

**Comment by nrc** [comment]
@eddyb as in, the `sign` field in the int lit types was always `Plus` (except with syntax extensions)?

**Comment by nrc** [comment]
In which case, this sounds like a good change, but I'd still like to see it batched with other changes since it is non-urgent.

**Comment by Manishearth** [comment]
Should we add some kind of github label for these PRs so it's easy to find them when we need to batch up? plugins-breaking or something

**Comment by pnkfelix** [comment]
@Manishearth 

I vote for "breaking-batch"

(Apologies to Walter White)

**Comment by steveklabnik** [comment]
This has been rebased, is it ready to go?

**Comment by Manishearth** [comment]
@oli-obk want to uplift this to the mega-breaking-batch PR? Since that will probably bitrot this when it lands.

**Comment by oli-obk** [comment]
> want to uplift this to the mega-breaking-batch PR? Since that will probably bitrot this when it lands.

already done

**Pull request closed without merging**